### PR TITLE
Tabs

### PR DESCRIPTION
**Pull request by dannyroberts**
Looking at the diff stats, it's kinda crazy that just the logic of the tabs in our side header is almost 2,000 lines of code. I think it'd be a lot easier to work with (and pare down) if we separated it out like this.

Interdependent with https://github.com/dimagi/commcarehq-prelogin/pull/72

### CODE REVIEW

File: corehq/tabs/uitab.py [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+                None,  # tab.org should be None for any non org page
+                view,
+                is_active,
+                request.couch_user.get_id,
+                get_language(),
+            ])
+            cache.delete(key)
+
+
+    @property
```
Comment (dimagimon) [inline corehq/tabs/uitab.py]: E303 too many blank lines (2)

**Comment by dannyroberts** [comment]
It's a shame that the diff doesn't treat deleting code from `corehq/apps/hqwebapp/models.py` and adding code to `corehq/tabs/tabclasses.py` as a rename. Here's the output of `git diff master:corehq/apps/hqwebapp/models.py TABS:corehq/tabs/tabclasses.py`: https://gist.github.com/dannyroberts/649a1b4aab20c627ea138546ee78afce.

And here's what gets left behind in `corehq/apps/hqwebapp/models.py`: https://github.com/dimagi/commcare-hq/blob/TABS/corehq/apps/hqwebapp/models.py.

**Comment by dannyroberts** [comment]
(still just adding a few things, but would like to hear from people if we're happy with this line of change)

**Comment by dannyroberts** [comment]
@emord buddy

**Comment by dannyroberts** [comment]
Now that staging is back up... I've checked that this doesn't destroy anything! I vote to get this in before it creates a conflict! Maybe @esoergel or @benrudolph can help with review?

**Comment by benrudolph** [comment]
woot looks good! :file_folder: :file_folder: :open_file_folder:

**Pull request merged**